utilities/cli/README.md

```diff
@@ -135,6 +135,196 @@ The new CLI introduces development lifecycle features:
+  - Can be a single string: `"--privileged --net=host"`
+  - Or an array: `["--privileged", "--net=host"]`
+  - **Note**: These arguments apply to both the container that builds your application and the container that runs it
+  - **Common use cases**: GPU access (`--gpus=all`), device access (`--device=/dev/video0`), privileged operations (`--privileged`)
+
+##### **Complete Example**
+
+```json
+{
+  "metadata": {
+    "modes": {
+
+      "default": {
+        "description": "Standard camera input for development and testing",
+        "requirements": ["camera", "model"],
+        "run": {
+          "command": "python3 <holohub_app_source>/app.py --source camera",
+          "workdir": "holohub_bin"
+        }
+      },
+
+      "production": {
+        "description": "High-performance mode with GPU acceleration",
+        "requirements": ["gpu", "model", "tensorrt"],
+        "build": {
+          "depends": ["tensorrt_backend", "gpu_ops"],
+          "docker_args": ["--build-arg", "TENSORRT_VERSION=8.6", "--network=host"],
+          "configure_args": ["-DUSE_TENSORRT=ON", "-DCUDA_ARCH=sm_86"]
+        },
+        "run": {
+          "command": "python3 <holohub_app_source>/app.py --backend tensorrt --optimization-level 3",
+          "workdir": "holohub_bin",
+          "docker_args": ["--gpus=all", "--shm-size=1g"]
+        }
+      },
+
+      "aja_hardware": {
+        "description": "AJA capture card with hardware overlay",
+        "requirements": ["aja_card_with_overlay", "model"],
+        "build": {
+          "depends": ["aja_source"]
+        },
+        "run": {
+          "command": "python3 <holohub_app_source>/app.py --source=aja --config=overlay.yaml",
+          "workdir": "holohub_bin",
+          "docker_args": [
+            "--privileged",
+            "--device=/dev/ajantv2",
+            "-v", "/dev:/dev"
+          ]
+        }
+      }
+    }
+  }
+}
+```
+
+##### **Advanced Pattern: Separate Build and Run Modes**
+
+For cases where build and run containers need different Docker configurations, you can create separate modes that share the same Docker image:
+
+```json
+{
+  "modes": {
+
+    "production_build": {
+      "description": "Build production image with network access",
+      "build": {
+        "depends": ["tensorrt_backend"],
+        "docker_args": ["--build-arg", "REGISTRY_TOKEN=xyz"]
+      },
+      "run": {
+        "docker_args": ["--network=host"],
+        "command": "echo 'Build complete'"
+      }
+    },
+
+    "production_run": {
+      "description": "Run production app with GPU access",
+      "run": {
+        "docker_args": ["--gpus=all", "--shm-size=1g"],
+        "command": "python3 <holohub_app_source>/app.py --gpu"
+      }
+    }
+  }
+}
+```
+
+**Usage:**
+```bash
+# Phase 1: Build with network access
+./holohub build myapp production_build
+
+# Phase 2: Run with GPU access (reuses image from phase 1)
+./holohub run myapp production_run --no-docker-build
```

Comment:
Is this new? was this design discussed?

Sorry but I see some big issues with this design:
- `run` instruction in `build` mode doesn't actually run an app in a certain mode, (here it does just echo)
- I can't tell what the behavior of the `build` instruction in `run` mode would look like, if it's even defined, but the concern is the same as above, and the example here requiring to add `--no-docker-build` illustrates that problem


run/build are instructions, I think conflating them with modes is a mistake. The designed we reviewed internally plans for future support of a separate run vs build environments attached to instructions, not to modes.

Reply:
I'll change `build.docker_args` to `build.docker_build_args` and `run.docker_args` to `run.docker_run_args`